datafusion/proto/Cargo.toml

```diff
@@ -55,5 +55,6 @@ doc-comment = "0.3"
 tokio = "1.18"
 
 [build-dependencies]
-pbjson-build = { version = "0.5" }
-prost-build = { version = "0.11.1" }
+# Pin these dependencies so that the generated output is deterministic
+pbjson-build = { version = "=0.5.1" }
```

Comment:
Shall we just make the version 0.5.1 (rather than pin it) so we automatically get updates?

Reply:
Minor version bumps can and will still change the code - see #3958

Reply:
Right -- I guess I was saying with this approach we won't see 0.5.2 if it is released.

If we left 

```
pbjson-build = { version = "0.5.1" }
```

We would start seeing local changes in our checkouts again potentially which might prompt someone to check them in?

I don't feel strongly

Reply:
> We would start seeing local changes in our checkouts again

As this wouldn't be correlated with someone making a change to the proto definitions I think this could be surprising. What would more likely happen is there local lockfile will hold back the upgrade, then they will get strange CI failures. If they then check in the proto files, people will then get local changes until their lockfile is updated... Basically a bit of a mess :sweat_smile: 

I don't like this either, checking in this code feels wrong, but I'm at a loss as to how to make this work better